tests/atoms/PutLinkUTest.cxxtest

```diff
@@ -123,3 +127,74 @@ void PutLinkUTest::test_filtering()
 
 	logger().info("END TEST: %s", __FUNCTION__);
 }
+
+void PutLinkUTest::test_lambda()
+{
+	logger().info("BEGIN TEST: %s", __FUNCTION__);
+
+	// Test that the substitution follows the order of variables
+	// provided in the lambda rather than left-to-right in the body.
+	Handle lam =
+		L(LAMBDA_LINK,
+		  L(VARIABLE_LIST,
+		    N(VARIABLE_NODE, "$x"),
+		    N(VARIABLE_NODE, "$y")),
+		  L(INHERITANCE_LINK,
+		    N(VARIABLE_NODE, "$y"),
+		    N(VARIABLE_NODE, "$x")));
+	Handle put =
+		L(PUT_LINK,
+		  lam,
+		  L(LIST_LINK,
+		    N(CONCEPT_NODE, "animal"),
+		    N(CONCEPT_NODE, "coyote")));
+
+	Instantiator inst(&_as);
+	Handle putted = inst.execute(put);
+
+	Handle expected_putted =
+		L(INHERITANCE_LINK,
+		  N(CONCEPT_NODE, "coyote"),
+		  N(CONCEPT_NODE, "animal"));
+
+	printf("Expecting %s\n", expected_putted->toString().c_str());
+	printf("Got %s\n", putted->toString().c_str());
+	TS_ASSERT_EQUALS(putted, expected_putted);
+
+	logger().info("END TEST: %s", __FUNCTION__);
+}
+
+void PutLinkUTest::test_forall()
+{
+	logger().info("BEGIN TEST: %s", __FUNCTION__);
+
+	// Identical to test_lambda but uses a forall instead.
+	Handle lam =
+		L(FORALL_LINK,
+		  L(VARIABLE_LIST,
+		    N(VARIABLE_NODE, "$x"),
+		    N(VARIABLE_NODE, "$y")),
+		  L(INHERITANCE_LINK,
+		    N(VARIABLE_NODE, "$y"),
+		    N(VARIABLE_NODE, "$x")));
+	Handle put =
+		L(PUT_LINK,
+		  lam,
+		  L(LIST_LINK,
+		    N(CONCEPT_NODE, "animal"),
+		    N(CONCEPT_NODE, "coyote")));
+
```

Comment:
@AmeBel I probably shouldn't insist on that cause it might bring more confusion than anything. On the practical level I agree with @linas. On a more philosophical level, I think one can indeed define a formal system where forall inherits the properties of a lambda while not falling into inconsistencies.